src/application/service/Wishlist.service.spec.ts

```diff
@@ -0,0 +1,163 @@
+import { ProductDbQueryPort } from '../port/in/product/ProductDbQueryPort';
+import { WishlistDbCommandPort } from '../port/out/WishlistDbCommandPort';
+import { WishlistDbQueryPort } from '../port/out/WishlistDbQueryPort';
+
+describe('WishlistService', () => {
+  let service: WishlistService;
+  let queryPortMock: jest.Mocked<WishlistDbQueryPort>;
+  let commandPortMock: jest.Mocked<WishlistDbCommandPort>;
+  let productPortMock: jest.Mocked<ProductDbQueryPort>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        WishlistService,
+        {
+          provide: 'WishlistGateway',
+          useValue: {
+            addWishlist: jest.fn(),
+            removeWishlist: jest.fn(),
+            getWishlistById: jest.fn(),
+            getAllByUserId: jest.fn(),
+            getByUserIdAndReceiverName: jest.fn(),
+          },
+        },
+        {
+          provide: 'ProductGateway',
+          useValue: {
+            getWishlistProductsByIds: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<WishlistService>(WishlistService);
+    queryPortMock = module.get<jest.Mocked<WishlistDbQueryPort>>('WishlistGateway');
+    commandPortMock = module.get<jest.Mocked<WishlistDbCommandPort>>('WishlistGateway');
+    productPortMock = module.get<jest.Mocked<ProductDbQueryPort>>('ProductGateway');
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('addWishlist', () => {
+    it('위시리스트에 상품을 추가할 수 있다', async () => {
+      //given
+      const command = createWishlistMock;
+
+      //when
+      await service.addWishlist(command);
+
+      //then
+      expect(commandPortMock.addWishlist).toHaveBeenCalledWith(command);
+    });
+  });
+
+  describe('removeWishlist', () => {
+    it('자신의 위시리스트에서 상품을 삭제할 수 있다', async () => {
+      //given
+      queryPortMock.getWishlistById.mockResolvedValue(domainWishlistMock);
+
+      //when
+      await service.removeWishlist(domainWishlistMock.id, domainWishlistMock.userId);
+
+      //then
+      expect(commandPortMock.removeWishlist).toHaveBeenCalledWith(domainWishlistMock.id);
+    });
+    it('존재하지 않는 위시리스트 ID일 경우 예외를 던진다', async () => {
+      // given
+      const wishlistId = 999;
+      const userId = 1;
+      commandPortMock.removeWishlist.mockImplementation(() => {
+        throw new WishlistNotFoundException();
+      });
+
+      // then
+      await expect(service.removeWishlist(wishlistId, userId)).rejects.toThrow(
+        WishlistNotFoundException,
+      );
+    });
+    it('다른 사용자의 위시리스트일 경우 예외를 던진다', async () => {
+      // given
+      queryPortMock.getWishlistById.mockResolvedValue({ ...domainWishlistMock, userId: 999 });
+
+      // then
+      await expect(service.removeWishlist(domainWishlistMock.id, 1)).rejects.toThrow(
+        ForbiddenWishlistAccessException,
+      );
+    });
+  });
+  describe('getWishlistGroups', () => {
```

Comment:
테스트 케이스에서 테스트 하고자 하는 목적이 드러나게 작성해주시면 좋을 것 같습니다.



```suggestion
    it('다른 사용자의 위시리스트일 경우 예외를 던진다', async () => {
      // given
      const wishlistOwnerId = 999;
      queryPortMock.getWishlistById.mockResolvedValue({ ...domainWishlistMock, userId: 999 });

      // when & then
      const requestUserId = 1;
      await expect(service.removeWishlist(domainWishlistMock.id, requestedUserId)).rejects.toThrow(
        ForbiddenWishlistAccessException,
      );
    });
  });
  
  describe('getWishlistGroups', () => {
```

Reply:
네 수정했습니다!